src/aosm/azext_aosm/cli_handlers/onboarding_base_handler.py

```diff
@@ -42,14 +43,21 @@ def output_folder_file_name(self) -> str:
     def __init__(self, input_json: str | None = None):
         # Config may be optional (to generate blank config file)
         input_json_path = Path(input_json) if input_json else None
-        config_dict = (
-            self._read_config_from_file(input_json_path) if input_json_path else {}
-        )
+        print(input_json_path)
         # Ensure config is of correct type
         try:
-            self.config = self._get_config(config_dict)
+            if input_json_path.suffix == '.jsonc':
+                config_dict = (
+                    self._read_config_from_file(input_json_path) if input_json_path else {}
+                )
+                self.config = self._get_input_config(config_dict)
+            elif input_json_path.suffix == '.json':
+                config_dict = self._read_params_config_from_file(input_json_path) if input_json_path else {}

```

Comment:
Do we ever need the `if input_json_path else {}` branch for `.json`?  
This only happens if we have the output folder, so `input_json_path` should never be false in this case - is that right?